pkg/hierarchy/manager.go

```diff
@@ -38,11 +38,31 @@ func NewManager[CQ clusterQueueNode[C], C cohortNode[CQ, C]](newCohort func(stri
 }
 
 func (m *Manager[CQ, C]) AddClusterQueue(cq CQ) {
-	m.ClusterQueues[cq.GetName()] = cq
+	m.clusterQueues[cq.GetName()] = cq
+}
+
+func (m *Manager[CQ, C]) GetClusterQueue(name string) CQ {
+	return m.clusterQueues[name]
+}
+
+func (m *Manager[CQ, C]) GetClusterQueueNames() []string {
+	clusterQueueNames := make([]string, 0, len(m.clusterQueues))
+	for k := range m.clusterQueues {
+		clusterQueueNames = append(clusterQueueNames, k)
+	}
+	return clusterQueueNames
+}
+
+func (m *Manager[CQ, C]) GetClusterQueuesCopy() map[string]CQ {
```

Comment:
```suggestion
func (m *Manager[CQ, C]) ClusterQueues() map[string]CQ {
```
I would prefer to think "Copy" is just an internal detail.